microk8s-resources/default-hooks/reconcile.d/10-cni-restart

```diff
@@ -1,38 +0,0 @@
-#!/bin/bash
-
-. "${SNAP}/actions/common/utils.sh"
-
-KUBECTL="${SNAP}/microk8s-kubectl.wrapper"
-
-# Calico reload deployment
-if ! [ -e "${SNAP_DATA}/var/lock/no-cni-reload" ] &&
-   [ -e "${SNAP_DATA}/args/cni-network/cni.yaml" ] &&
-   [ -e "${SNAP_DATA}/var/lock/ha-cluster" ] &&
-   [ -e "${SNAP_DATA}/var/lock/cni-loaded" ] &&
-   [ -e "${SNAP_DATA}/var/lock/no-flanneld" ] &&
-   [ -e "${SNAP_DATA}/var/lock/cni-needs-reload" ]
-then
-
-  echo "Reloading the calico CNI"
-  if (is_apiserver_ready)  &&
-      "${KUBECTL}" --request-timeout 2m describe -n kube-system daemonset.apps/calico-node &&
-      "${KUBECTL}" --request-timeout 2m describe -n kube-system deployment.apps/calico-kube-controllers &&
-      "${KUBECTL}" --request-timeout 2m rollout restart -n kube-system daemonset.apps/calico-node &&
-      "${KUBECTL}" --request-timeout 2m rollout restart -n kube-system deployment.apps/calico-kube-controllers
-  then
-    rm "${SNAP_DATA}/var/lock/cni-needs-reload"
-  fi
-fi
-
-if  ! [ -e "${SNAP_DATA}/var/lock/skip-offloading" ] &&
-    ! [ -e "${SNAP_DATA}/var/lock/offloading-patch-applied" ] &&
-      [ -e "${SNAP_DATA}/args/cni-network/cni.yaml" ] &&
-      [ -e "${SNAP_DATA}/var/lock/ha-cluster" ] &&
-      [ -e "${SNAP_DATA}/var/lock/cni-loaded" ] &&
-      [ -e "${SNAP_DATA}/var/lock/no-flanneld" ] &&
-    ! grep -q "ChecksumOffloadBroken" "${SNAP_DATA}/args/cni-network/cni.yaml" &&
-    "${KUBECTL}" patch felixconfigurations default --patch '{"spec":{"featureDetectOverride":"ChecksumOffloadBroken=true"}}' --type=merge
```

Comment:
This patching is not needed anymore?

Reply:
Not anymore, for some releases now this has been handled by the Calico manifest.

Reply:
Did we force deployments to move to the calico versions that address this issue?

Reply:
yes. deployments from 1.26 onwards have the right cni.yaml manifest. for deployments starting from older versions, this hook has run already while upgrading and can be therefore safely removed